enrichment-definition/src/main/java/eu/europeana/enrichment/model/impl/TopicEntityImpl.java

```diff
@@ -0,0 +1,219 @@
+	// language based description
+	private Map<String,String> descriptions;
+	
+	
+	private List<TermEntity> terms;
+	
+	private List<TermEntity> keywords;
+	
+	private String modelId;
+	
+	private Date created;
+	private Date modified;
+	
+	
+	public TopicEntityImpl()
+	{
+		
+	}
+
+	
+
+	public TopicEntityImpl(TopicEntity topicEntity) {
+		this.topicID = topicEntity.getTopicID();
+		this.identifier = topicEntity.getIdentifier();
+		this.labels = topicEntity.getLabel();
+		this.descriptions = topicEntity.getDescription();
+		this.terms = topicEntity.getTopicTerms();
+		this.keywords = topicEntity.getTopicKeywords();
+		this.model = topicEntity.getTopicModel();
+		this.modelId = topicEntity.getModelId();
+		this.created = topicEntity.getCreatedDate();
+		if (topicEntity.getModifiedDate()==null)
+			this.modified = new Date();
+		else
+			this.modified = topicEntity.getModifiedDate();
+	}
+
+	@Override
+	public String getTopicID() {
+		return this.topicID;
+	}
+
+	@Override
+	public void setTopicID(String id) {
+		this.topicID = id;
+
+	}
+
+	@Override
+	public String getIdentifier() {
+		return this.identifier;
+	}
+
+	@Override
+	public void setIdentifier(String identifier) {
+		this.identifier = identifier;
+
+	}
+
+	@Override
+	public List<String> getLabel() {
+		return this.labels;
+	}
+
+	@Override
+	public void setLabel(List<String> label) {
+		this.labels = label;
+	}
+
+	@Override
+	public Map<String, String> getDescription() {
+		return this.descriptions;
+	}
+
+	@Override
+	public void setDescription(Map<String, String> descr) {
+		this.descriptions = descr;
+	}
+
+	@Override
+	public List<TermEntity> getTopicTerms() {
+		return this.terms;
+	}
+
+	@Override
+	public void setTopicTerms(List<TermEntity> terms) {
+		this.terms = terms;
+	}
+
+	@Override
+	public List<TermEntity> getTopicKeywords() {
+		return this.keywords;
+	}
+
+	@Override
+	public void setTopicKeywords(List<TermEntity> keywords) {
+		this.keywords = keywords;
+	}
+
+	@Override
+	public TopicModel getTopicModel() {
+		return this.model;
+	}
+
+	@Override
+	public void setTopicModel(TopicModel model) {
+		this.model = model;
+
+	}
+
+	@Override
+	public Date getCreatedDate() {
+		return this.created;
+	}
+
+	@Override
+	public Date getModifiedDate() {
+		return this.modified;
+	}
+
+	@Override
+	public void setCreatedDate(Date date) {
+		this.created = date;
+	}
+
+	@Override
+	public void setModifiedDate(Date date) {
+		this.modified = date;
+	}
+	
+	@Override
+	public String getId() {
+		return _id;
+	}
+
+	@Override
+	public String toJSON() throws JsonProcessingException {
+		ObjectMapper mapper = new ObjectMapper();
+		
+		return mapper.writeValueAsString(this);
+	}
+
+	@Override
+	public String getModelId() {
+		return modelId;
+	}
+
+	@Override
+	public int hashCode() {
+		return Objects.hash(_id, created, descriptions, identifier, keywords, labels, model, modelId, modified, terms,
+				topicID);
+	}
+
+	@Override
+	public boolean equals(Object obj) {
```

Comment:
As far as I can see, we do not use the equal operator for the TopicEntityImpl, so please remove this methos as well as the hashCode one above.

Reply:
it is implicitly used that's why I added them; would not have created them otherwise

Reply:
we should comment it out for the time being, probably we will have an update in the specs

Reply:
changed to using only identifier for equals; this method is still needed when deleting topics